kofta/src/app/components/add-to-calendar/AddToCalendar.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useMemo, useRef, useState } from "react";
+
+import makeUrls, { CalendarEvent } from "./makeUrls";
+
+type CalendarURLs = ReturnType<typeof makeUrls>;
+
+const useAutoFocus = () => {
+  const elementRef = useRef<HTMLElement>(null);
+
+  useEffect(
+    () => {
+      const previous = document.activeElement;
+      const element = elementRef.current;
+
+      if (element) {
+        element.focus();
+      }
+
+      if (previous instanceof HTMLElement) {
+        return () => previous.focus();
+      }
+
+      return undefined;
+    },
+    []
+  );
+
+  return elementRef;
+};
+
+type OpenStateToggle = (event?: React.MouseEvent) => void;
+
+const useOpenState = (initialOpen: boolean): [boolean, OpenStateToggle] => {
+  const [open, setOpen] = useState<boolean>(initialOpen);
+  const onToggle = () => setOpen(current => !current);
+
+  useEffect(
+    () => {
+      if (open) {
+        const onClose = () => setOpen(false);
+        document.addEventListener("click", onClose);
+
+        return () => document.removeEventListener("click", onClose);
+      }
+
+      return undefined;
+    },
+    [open, setOpen]
+  );
+
+  return [open, onToggle];
+};
+
+type CalendarRef = HTMLAnchorElement;
+type CalendarProps = {
+  children: React.ReactNode;
+  filename?: string;
+  href: string;
+};
+
+const Calendar = React.forwardRef<CalendarRef, CalendarProps>(
+  ({ children, filename = false, href }, ref) => (
+    <a ref={ref} download={filename} href={href} target="_blank" rel="noopener noreferrer">
+      {children}
+    </a>
+  )
+);
+
+type DropdownProps = {
+  filename: string;
+  onToggle: OpenStateToggle;
+  urls: CalendarURLs;
+};
+
+const Dropdown: React.FC<DropdownProps> = ({ filename, onToggle, urls }) => {
+  const ref = useAutoFocus() as React.RefObject<HTMLAnchorElement>;
```

Comment:
why do you need to focus this?

Reply:
Clicking the dropdown and pressing "enter" on keyboard allows you to download the ref item.

Open dropdown, and click enter => downloads Apple Calendar .ics
` <Calendar href={urls.ics} filename={filename} ref={ref}>
        Apple Calendar
      </Calendar>`

Open dropdown, and click enter => opens Google Calendar in new tab
`<Calendar href={urls.google} ref={ref}>
        Google
      </Calendar>`

Removing this will yield no action upon pressing "enter".